Ingress/controllers/haproxy/README.md

```diff
@@ -0,0 +1,310 @@
++--------------------+
+| Database           |
++--------------------+
+| information_schema |
+| mysql              |
+| performance_schema |
++--------------------+
+```
+
+
+#### Cross-cluster loadbalancing
+
+First setup your 2 clusters, and a kubeconfig secret as described in the [sharing clusters example] (../../examples/sharing-clusters/README.md). We will create a loadbalancer in our first cluster (US) and have it publish the services from the second cluster (EU). This is the entire modified loadbalancer manifest:
+
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: service-loadbalancer
+  labels:
+    app: service-loadbalancer
+    version: v1
+spec:
+  replicas: 1
+  selector:
+    app: service-loadbalancer
+    version: v1
+  template:
+    metadata:
+      labels:
+        app: service-loadbalancer
+        version: v1
+    spec:
+      volumes:
+      # token from the eu cluster, must already exist
+      # and match the name of the volume using in container
+      - name: eu-config
+        secret:
+          secretName: kubeconfig
+      nodeSelector:
+        role: loadbalancer
+      containers:
+      - image: gcr.io/google_containers/hlbc:0.1
+        imagePullPolicy: Always
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+            scheme: HTTP
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+        name: haproxy
+        ports:
+        # All http services
+        - containerPort: 80
+          hostPort: 80
+          protocol: TCP
+        # nginx https
+        - containerPort: 443
+          hostPort: 8080
+          protocol: TCP
+        # mysql
+        - containerPort: 3306
+          hostPort: 3306
+          protocol: TCP
+        # haproxy stats
+        - containerPort: 1936
+          hostPort: 1936
+          protocol: TCP
+        resources: {}
+        args:
+        - --tcp-services=mysql:3306,nginxsvc:443
+        - --use-kubernetes-cluster-service=false
+        # use-kubernetes-cluster-service=false in conjunction with the
+        # kube/config will force the service-loadbalancer to watch for
+        # services form the eu cluster.
+        volumeMounts:
+        - mountPath: /.kube
+          name: eu-config
+        env:
+        - name: KUBECONFIG
+          value: /.kube/config
+```
+
+Note that it is essentially the same as the rc.yaml checked into the service-loadbalancer directory expect that it consumes the kubeconfig secret as an extra KUBECONFIG environment variable.
+
+```cmd
+$ kubectl config use-context <us-clustername>
+$ kubectl create -f rc.yaml
+$ kubectl get pods -o wide
+service-loadbalancer-5o2p4   1/1       Running   0          13m       kubernetes-minion-5jtd
+$ kubectl get node kubernetes-minion-5jtd -o json | grep -i externalip -A 2
+                "type": "ExternalIP",
+                "address": "104.197.81.116"
+$ curl http://104.197.81.116/nginxsvc
+Europe
+```
+
```

Comment:
Suggest adding a small section on Ingress here if you have time. Just try creating a couple of Ingress objects (as described in https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/user-guide/ingress.md) and write down what to expect. 

Also suggest adding a section on the new Service annotation support we added.